components/shared/webrender/rendering_context.rs

```diff
@@ -9,17 +9,39 @@ use std::ffi::c_void;
 use std::rc::Rc;
 
 use euclid::default::Size2D;
+use gleam::gl;
+use log::warn;
 use surfman::chains::{PreserveBuffer, SwapChain};
 use surfman::{
     Adapter, Connection, Context, ContextAttributeFlags, ContextAttributes, Device, Error, GLApi,
     GLVersion, NativeContext, NativeDevice, NativeWidget, Surface, SurfaceAccess, SurfaceInfo,
     SurfaceTexture, SurfaceType,
 };
 
+pub trait RenderingContext {
+    fn device(&self) -> NativeDevice;
+    fn context(&self) -> NativeContext;
+    fn resize(&self, size: Size2D<i32>);
+    fn present(&self);
+    fn bind_native_surface_to_context(&self, native_widget: NativeWidget);
+    fn connection(&self) -> Connection;
+    fn adapter(&self) -> Adapter;
+    fn make_current(&self);
+    fn framebuffer_object(&self) -> u32;
+    fn gl_api(&self) -> Rc<dyn gleam::gl::Gl>;
+    fn gl_version(&self) -> GLVersion;
+    fn invalidate_native_surface(&self);
+    fn replace_native_surface(
+        &self,
+        native_widget: *mut c_void,
+        coords: euclid::Size2D<i32, webrender_api::units::DevicePixel>,
+    );
+}
```

Comment:
Could you add the documentation for this trait and its methods? You can take #34888 for some reference.